### ISSUE TITLE
Allow to set the cmake build flags

### ISSUE THREAD

**Issue by Hugal31** [colcon/colcon-cmake#141]
For now, Colcon automatically pass  `-j{os.cpu_count()} -l{os.cpu_count}` to the `cmake --build` command, unless the MAKEFLAGS environment variable contains them. There is no other way to control what arguments the builder receives. This works fine with make, but is useless with Ninja, which [doesn't use an environment variable](https://github.com/ninja-build/ninja/issues/797).

As a workaround, I use taskset to change the CPU affinity and limit the number of parallel build, but it only limit the number of parallelism and thus doesn't work with tools such as `distcc` which allow to build with more parallelism.

It would be great to at least be able to control the parallelism via a colcon flag, or perhaps better, be able to directly pass arguments to the `cmake --build` invocation.

**Comment by andrejpan** [comment]
I am a bit new in the colcon world, but as of today, colcon so far does not support distributed compilation with distcc?

**Comment by Hugal31** [comment]
Yes it does, but it uses the number of cpu of the host machine as parallelism by default. Or it let GNU Make uses the MAKEFLAGS environment variable, but it only works with GNU Make and not Ninja.

**Comment by andrejpan** [comment]
Do you perhaps have a readme on how did you set up your environment?

**Comment by Hugal31** [comment]
No. I don't personally use distcc, I just want to be able to run colcon without having it using all my cores. My PR make it so I can run `colcon build --executor sequential --cmake-build-args ' --parallel' 6` for example.

However my colleagues use distcc. There's nothing particular about it with colcon. They just use something `--executor sequential` and `--cmake-build-args ' --parallel' $(distcc -j)`
